keps/sig-storage/695-skip-permission-change/README.md

```diff
@@ -0,0 +1,287 @@
+ - `OnRootMismatch` --> Only perform permission and ownership change if permissions of top level directory does not match with expected permissions and ownership.
+
+```go
+type PodFSGroupChangePolicy string
+
+const(
+    OnRootMismatch PodFSGroupChangePolicy = "OnRootMismatch"
+    AlwaysChangeVolumePermission PodFSGroupChangePolicy = "Always"
+)
+
+type PodSecurityContext struct {
+    // FSGroupChangePolicy ← new field
+    // Defines behavior of changing ownership and permission of the volume
+    // before being exposed inside Pod. This field will only apply to
+    // volume types which support fsGroup based ownership(and permissions).
+    // It will have no effect on ephemeral volume types such as: secret, configmaps
+    // and emptydir.
+    // + optional
+    FSGroupChangePolicy *PodFSGroupChangePolicy
+}
+```
+
+### Risks and Mitigations
+
+- One of the risks is if user volume's permission was previously changed using old algorithm(which changes permission of top level directory first) and user opts in for `OnRootMismatch` `FSGroupChangePolicy` then we can't distinguish if the volume was previously only partially recursively chown'd.
+
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate
+    - Feature gate name: ConfigurableFSGroupPolicy
+    - Components depending on the feature gate: kubelet, kube-apiserver
+
+* **Does enabling the feature change any default behavior?**
+  No enabling the feature gate does not change any default behaviour.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Yes the feature-gate can be disabled once enabled. This will cause all volume mounts that
+  require permission change to use recursive permission and ownership change. This is the default
+  behaviour without the feature gate.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  For pods that have expected value in `pod.Spec.SecurityContext.FSGroupChangePolicy` as defined in this KEP,
+  it will start using specified policy.
+
+* **Are there any tests for feature enablement/disablement?**
+  There aren't any e2e but there are unit tests that cover this behaviour.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Since this feature requires users to opt-in by setting new field in pod spec, it should
+  not impact already running workloads.
+
+* **What specific metrics should inform a rollback?**
+  If after enabling this feature and setting `pod.Spec.SecurityContext.FSGroupChangePolicy`
+  users notice an increase in volume mount time via `storage_operation_duration_seconds{operation_name=volume_mount}`
+  or an increase in mount error count via `storage_operation_errors_total{operation_name=volume_mount}`
+  then a cluster-admin may want to rollback the feature.
```

Comment:
Will enabling this feature cause an increase in volume mount time?  I thought it should be the opposite?

Reply:
you are right but this is just something admin can use to track down health of volume mount operation. If mount timings have substantially increased somehow after this feature then something is broken!